frontend/public/components/utils/value-from-pair.jsx

```diff
@@ -45,25 +130,47 @@ const ResourceFieldRef = ({data: {containerName, resource}}) => <span>
 
 const keyStringToComponent = {
   fieldRef: FieldRef,
-  secretKeyRef: SecretKeyRef,
-  configMapKeyRef: ConfigMapKeyRef,
+  secretKeyRef: {
+    component: ConfigMapOrSecretKeyRef,
+    kind: 'Secret'
+  },
+  configMapKeyRef: {
+    component: ConfigMapOrSecretKeyRef,
+    kind: 'ConfigMap'
+  },
   resourceFieldRef: ResourceFieldRef,
 };
 
 export class ValueFromPair extends React.PureComponent {
   constructor (props) {
     super(props);
 
-    this.valueFromKey = Object.keys(this.props.pair)[0];
-    this.Component = keyStringToComponent[this.valueFromKey];
+    this.onChangeVal = (...args) => this._onChangeVal(...args);
+  }
+
+  _onChangeVal(value) {
+    const {onChange} = this.props;
+    const e = {
+      'target': {
+        'value': value
+      }
+    };
```

Comment:
const e = { target: { value } };

Reply:
I don't love creating fake events. What is this doing?

Reply:
Changed to use shorthand. 

This is just formatting the data similar to what is expected in the PairElement onChangeValue method and passing it back to it using props.